schema/validator.go

```diff
@@ -73,3 +88,28 @@ type unimplemented string
 func (v unimplemented) Validate(src io.Reader) error {
 	return fmt.Errorf("%s: unimplemented", v)
 }
+
+func validateManifestDescendants(r io.Reader) error {
+	header := v1.Manifest{}
+
+	buf, err := ioutil.ReadAll(r)
+	if err != nil {
+		return errors.Wrapf(err, "error reading the io stream")
+	}
+
+	err = json.Unmarshal(buf, &header)
+	if err != nil {
+		return errors.Wrap(err, "manifest format mismatch")
+	}
+
+	if header.Config.MediaType != string(v1.MediaTypeImageConfig) {
+		fmt.Printf("warning: config %s has a non OCI media type: %s\n", header.Config.Digest, header.Config.MediaType)
```

Comment:
The problem isn't that it has a non-OCI type (`application/vnd.oci.image.manifest.list.v1+json` is an OCI type, and it deserves this warning).  My [suggested wording is][1]:

> {blobDigest} {property} has an unknown media type: {type}

That's a bit complicated with your APIs, because you take a reader and not a descriptor or digest.  In my mock-up, the validator takes [a digest and CAS engine][2], so it's easy to make messages like that.  But it makes sense to want a way to validate a reader without having to inject it into a CAS engine, and for that initial reader-based validation I think the message should just be:

> {blobDigest} {property} has an unknown media type: {type}

For this line here, that's:

    fmt.Println("warning: config has an unknown media type: %s", header.Config.MediaType)

[1]: https://github.com/opencontainers/image-spec/pull/341#discussion_r82114804
[2]: https://github.com/wking/image-tools/commit/4f0b3c1bae97d212680dc73ee8b35b6e47b8a8c6#diff-8351286f57eb81e245c9c99c07f3b34fR27